src/Sylius/Component/Core/Currency/ChannelAwareCurrencyProvider.php

```diff
@@ -11,29 +11,32 @@
 
 namespace Sylius\Component\Core\Currency;
 
+use Sylius\Bundle\ResourceBundle\Doctrine\ORM\EntityRepository;
 use Sylius\Component\Channel\Context\ChannelContextInterface;
 use Sylius\Component\Currency\Model\CurrencyInterface;
-use Sylius\Component\Currency\Provider\CurrencyProviderInterface;
+use Sylius\Component\Currency\Provider\CurrencyProvider;
 
 /**
  * Currency provider, which returns currencies enabled for this channel.
  *
  * @author Paweł Jędrzejewski <pawel@sylius.org>
+ * @author Fernando Caraballo Ortiz <caraballo.ortiz@gmail.com.org>
  */
-class ChannelAwareCurrencyProvider implements CurrencyProviderInterface
+class ChannelAwareCurrencyProvider extends CurrencyProvider
 {
     /**
-     * Channel context.
-     *
      * @var ChannelContextInterface
      */
     protected $channelContext;
 
     /**
      * @param ChannelContextInterface $channelContext
+     * @param EntityRepository $currencyRepository
      */
-    public function __construct(ChannelContextInterface $channelContext)
+    public function __construct(ChannelContextInterface $channelContext, EntityRepository $currencyRepository)
```

Comment:
This should be `Sylius\Component\Resource\Repository\RepositoryInterface` to match the parent and allow for non-ORM repositories to be injected.